docs/quickstart/reproducibility.md

```diff
@@ -0,0 +1,45 @@
+---
+layout: default 
+title: Tutorial – Reproducibility
+description: In this section we will explore how to use lakeFS for reproducibility of data.
+parent: Quickstart
+nav_order: 80
+has_children: false
+---
+
+## The Benefits of Reproducible Data
+
+Data changes frequently. This makes the task of keeping track of its exact state over time difficult. Often people maintain only one state of their data––its current state.
+
+The negative effects of this are numerous. It becomes hard to:
+* Debug a data issue.
+* Validate machine learning training accuracy. (re-running a model over different data gives different results)
+* Comply with data audits.
+
+In comparison, lakeFS exposes a git-like interface to data that makes it possible to keep track of more than just the current state of data. This makes reproducing its state at any point in time straightforward.
+
+## Achieving Reproducibility with lakeFS
+
+To make data reproducible, we recommend taking a new commit of your lakeFS repository every time the data in it changes. As long as there’s a commit taken, the process to reproduce a given state is as simple as reading the data from a path that includes the unique `commit_id` generated for each commit.
+
+To read data at it’s current state, we can use a static path containing the repository and branch names. To give an example, if you have a repository named `example` with a branch named `main`, reading the latest state of this data into a Spark Dataframe is always:
+
+```bash
+df = spark.read.parquet(‘s3://example/main/”)
+```
+
+In a lakeFS repository, we are capable of taking many commits over the data, making many points in time reproducible. 
+
+![Commit History]({{ site.baseurl }}/assets/img/reproduce-commit-history.png)
+
+In the repository shown above, a new commit is taken each time a model training script is run, and the commit message includes the specific run number. 
+
+If we wanted to re-run the model training script and reproduce the exact same results for a historical run, say run #435, we could copy the commit ID associated with the run and read the data into a dataframe like so:
+
+```bash
+df = spark.read.parquet("s3://example/296e54fbee5e176f3f4f4aeb7e087f9d57515750e8c3d033b8b841778613cb23”)
```

Comment:
To read an object version at the time of a certain commit, users need to use commit as a ref, and replace the branch name with the commit id in the object patf. e.g.: 
val df2 = spark.read.parquet("s3a://my-repo/e2673a54a27d1a23bf4f71d13aec921bbe663c03211d96e518975a77347725ed/users.parquet")

Reply:
Similarly, this also does work without the parquet suffix. I prefer not to reference specific files.. ok with you?

Reply:
No.. I think this is very confusing.. why do you prefer not to give an example that interacts with a specific object?

Reply:
in reality, most times you want to read in a dataset it is a collection like "all orders at company X" and it involves many files that are all in the same format. 

Ideally you should never really be interacting with an individual file. it makes lakeFS seem limiting when you are pointing to and reading specific files.

Reply:
Good point, I see what you mean. 
Here is what I suggest - 
Can you use a path to an explicit data set? e.g. 
df = spark.read.parquet("s3://example/296e54fbee5e176f3f4f4aeb7e087f9d57515750e8c3d033b8b841778613cb23/orders”)

I also validated it works. My point is that it is confusing to point to a branch root. WDYT?

Reply:
I like it! I don't think we've shown an example like that before and it's a good one.